### ISSUE TITLE
Can't Compile

### ISSUE THREAD

**Issue by muellerryan** [dalerank/imspinner#3]
Hi - Awesome work on these spinners.

Please bear with me, as I have been mostly developing with embedded style (mostly Arduino) c/c++ programming and have just started getting into desktop development and most recently into dear imgui.  

I've been able to create an imgui environment with imgui, implot, imgui-knobs, etc in VS2022 on Win10.  It compiles and runs just fine. Once I include imspinner.h, it is giving me tons of "identifier not found", "undeclared identifier", etc.  

Here is my build output:
`Build started...
1>------ Build started: Project: imgui-implot-knobs-template, Configuration: Debug x64 ------
1>imgui-implot-knobs-template.cpp
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(51,5): error C3861: 'IM_PI': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(51,5): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(60,9): error C2065: 'ImGuiWindow': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(60,22): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(60,38): error C2039: 'GetCurrentWindow': is not a member of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(3226): message : see declaration of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(60,54): error C3861: 'GetCurrentWindow': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(61,13): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(64,28): error C2065: 'GImGui': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(65,36): error C2027: use of undefined type 'ImGuiContext'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(149): message : see declaration of 'ImGuiContext'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(65,33): error C2530: 'style': references must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(66,28): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(66,23): error C2737: 'id': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(68,15): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(71,22): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(71,22): error C2146: syntax error: missing ';' before identifier 'bb'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(71,22): error C3861: 'bb': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(72,16): error C2039: 'ItemSize': is not a member of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(3226): message : see declaration of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(72,25): error C2065: 'bb': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(72,24): error C3861: 'ItemSize': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(74,18): error C2065: 'bb': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(75,21): error C2039: 'ItemAdd': is not a member of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(3226): message : see declaration of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(75,29): error C2065: 'bb': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(75,12): error C3861: 'ItemAdd': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(113,9): error C2065: 'ImGuiWindow': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(113,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(113,9): error C2039: 'GetCurrentWindow': is not a member of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(3226): message : see declaration of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(113,9): error C3861: 'GetCurrentWindow': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(116,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(117,37): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(117,22): error C2737: 'num_segments': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(118,29): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(118,23): error C3861: 'ImAbs': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(120,29): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(120,21): error C2737: 'a_min': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(121,29): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(121,21): error C2737: 'a_max': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(126,13): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(126,60): error C3861: 'ImCos': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(127,53): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(129,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(132,172): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(134,9): error C2065: 'ImGuiWindow': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(134,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(134,9): error C2039: 'GetCurrentWindow': is not a member of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(3226): message : see declaration of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(134,9): error C3861: 'GetCurrentWindow': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(137,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(138,37): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(138,22): error C2737: 'num_segments': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(140,39): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(140,21): error C2737: 'bg_angle_offset': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(144,13): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(144,60): error C3861: 'ImCos': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(144,90): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(146,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(148,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(153,13): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(153,60): error C3861: 'ImCos': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(153,90): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(155,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(165,7): error C2065: 'ImGuiWindow': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(165,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(165,7): error C2039: 'GetCurrentWindow': is not a member of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(3226): message : see declaration of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(165,7): error C3861: 'GetCurrentWindow': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(168,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(169,35): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(169,20): error C2737: 'num_segments': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(171,37): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(171,19): error C2737: 'bg_angle_offset': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(175,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(175,56): error C3861: 'ImCos': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(175,86): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(177,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(179,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(179,59): error C3861: 'ImCos': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(179,93): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(180,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(180,59): error C3861: 'ImCos': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(180,106): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(185,7): error C2065: 'ImGuiWindow': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(185,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(185,7): error C2039: 'GetCurrentWindow': is not a member of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imgui\imgui.h(3226): message : see declaration of 'ImGui'
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(185,7): error C3861: 'GetCurrentWindow': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(187,31): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(188,33): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(188,21): error C2737: 'radiusbId': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(192,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(193,35): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(193,20): error C2737: 'num_segments': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(195,37): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(195,19): error C2737: 'bg_angle_offset': const object must be initialized
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(197,37): error C2065: 'IM_PI': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(197,23): error C3861: 'ImFmod': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(198,24): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(203,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(203,56): error C3861: 'ImCos': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(203,87): error C3861: 'ImSin': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(205,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(210,36): error C3861: 'ImMax': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(210,20): error C3861: 'ImMax': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(218,36): error C3861: 'ImMax': identifier not found
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(219,7): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(223,9): error C2065: 'window': undeclared identifier
1>C:\VSCode_git_repos\imgui-implot-knobs-template-vs\imspinner\imspinner.h(223,15): fatal  error C1003: error count exceeds 100; stopping compilation
1>Done building project "imgui-implot-knobs-template.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========`

For reference, I am using the imgui-docker branch, the master branch of implot, and the master branch of imgui-knobs.  This is using a very slightly modified version of the directx12 example code (to add in implot and imgui-knobs and of course imspinner) from imgui. 

Am I including the header wrong, do I have to add something to this header or imgui-implot-knobs-template.cpp to make it work?  I feel like I am missing some small #includes or something of that nature since I am getting so many errors.

So far I have tried including "imgui.h" to this header and while there are less errors, I am still getting the majority of the ones I pasted above. 

Please let me know what further information you need and what I can provide for you so that I can get a push in the right direction in learning how this library interfaces with imgui.  

Thank you,
Ryan

**Comment by dalerank** [comment]
seems u need imgui_internal.h before imspinner.h

**Comment by muellerryan** [comment]
Edit: I am an idiot, I had some superfluous testing code that was bugging it out - everything works on all example codes now. Thank you again!

Wow, thank you, that worked ~~for compiling~~.  

I am able to run the code with just the inclusion on the barebones imgui example code, ~~however I get an error when using the code I modified to work with implot, knobs, etc.~~

I will mark this closed since I was able to successfully compile ~~regardless of the ability to run, but I will include the error code in case you have a quick tip on it:~~

~~`C:\VSCode_git_repos\imgui-implot-knobs-template-vs\x64\Debug\imgui-implot-knobs-template.exe (process 19592) exited with code -1073741819.
Press any key to close this window . . .`~~
